### PR TITLE
RDoc-2640 [Python] Client API > Session > Cluster transaction > Compare Exchange [Replace C# samples]

### PR DESCRIPTION
**Pull request by reebhub**
Related issue:
[RDoc-2640](https://issues.hibernatingrhinos.com/issue/RDoc-2640/Python-Client-API-Session-Cluster-transaction-Compare-Exchange-Replace-C-samples)

### CODE REVIEW

File: Documentation/5.4/Raven.Documentation.Pages/client-api/session/cluster-transaction/compare-exchange.python.markdown [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+{PANEL: Create compare-exchange}
+
+{NOTE: }
+__Example__
+
+{CODE:python new_compare_exchange_sync@ClientApi\Session\ClusterTransaction\CompareExchange.py /}
+
+* `save_changes()` throws a `ConcurrencyException` if the key already exists.
+* An `InvalidOperationException` exception is thrown if the session was Not opened as __cluster-wide__.
```
Comment (poissoncorp) [inline Documentation/5.4/Raven.Documentation.Pages/client-api/session/cluster-transaction/compare-exchange.python.markdown:29]: RuntimeException

Comment (poissoncorp) [inline Documentation/5.4/Raven.Documentation.Pages/client-api/session/cluster-transaction/compare-exchange.python.markdown:29]:
```suggestion
* An `RuntimeException` exception is thrown if the session was Not opened as __cluster-wide__.
```

File: Documentation/5.4/Raven.Documentation.Pages/client-api/session/cluster-transaction/compare-exchange.python.markdown [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+{NOTE/}
+
+{NOTE: }
+__The CompareExchangeValue__
+
+| Parameters   | Type     | Description                                                        |
+|--------------|----------|--------------------------------------------------------------------|
+| **key**      | `str`    | The compare-exchange item key. This string can be up to 512 bytes. |
+| **value**    | `T`      | The value associated with the key                                  |
+| **index**    | `long`   | Index for concurrency control                                      |
```
Comment (poissoncorp) [inline Documentation/5.4/Raven.Documentation.Pages/client-api/session/cluster-transaction/compare-exchange.python.markdown:55]: i think that this description is too short - we should expand it a bit to describe why would we use it

File: Documentation/5.4/Raven.Documentation.Pages/client-api/session/cluster-transaction/compare-exchange.python.markdown [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+{NOTE: }
+__Get compare-exchange lazily__
+
+{CODE:python methods_sync_lazy_1@ClientApi\Session\ClusterTransaction\CompareExchange.py /}
+
+| Parameters  | Type       | Description               |
+|-------------|------------|---------------------------|
+| **key**     | `str`      | The key to retrieve       |
+| **keys**    | `str[]`    | Array of keys to retrieve |
```
Comment (poissoncorp) [inline Documentation/5.4/Raven.Documentation.Pages/client-api/session/cluster-transaction/compare-exchange.python.markdown:99]:
```suggestion
| **keys**    | `List[str]`    | Array of keys to retrieve |
```

File: Documentation/5.4/Raven.Documentation.Pages/client-api/session/cluster-transaction/compare-exchange.python.markdown [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+{PANEL: Delete compare-exchange}
+
+{NOTE: }
+
+{CODE:python methods_4_sync@ClientApi\Session\ClusterTransaction\CompareExchange.py /}
+
+| Parameters | Type                      | Description                                    |
+|------------|---------------------------|------------------------------------------------|
+| **key**    | `str`                     | The key of the compare-exchange item to delete |
+| **index**  | `long`                    | The index of this compare-exchange item        |
```
Comment (poissoncorp) [inline Documentation/5.4/Raven.Documentation.Pages/client-api/session/cluster-transaction/compare-exchange.python.markdown:117]: there are no `long` types in Python
the only numeric type is `int`

```suggestion
| **index**  | `int`                    | The index of this compare-exchange item        |
```

File: Documentation/5.4/Raven.Documentation.Pages/client-api/session/cluster-transaction/compare-exchange.python.markdown [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+
+{CODE:python methods_2_sync@ClientApi\Session\ClusterTransaction\CompareExchange.py /}
+
+| Parameters   | Type       | Description               |
+|--------------|------------|---------------------------|
+| **keys**     | `str[]`    | Array of keys to retrieve |
+
+| Return Value | Description |
+| ------------- | ----- |
+| `Dict<str, CompareExchangeValue<T>>` | If a key doesn't exist the associated value will be `None` |
```
Comment (poissoncorp) [inline Documentation/5.4/Raven.Documentation.Pages/client-api/session/cluster-transaction/compare-exchange.python.markdown:88]: note:
in Python everyone is used to something more like this:
```suggestion
| `Dict[str, CompareExchangeValue[T]]` | If a key doesn't exist the associated value will be `None` |
```
but it's a small difference - you can change it if you want to increase the quality

File: Documentation/5.4/Raven.Documentation.Pages/client-api/session/cluster-transaction/compare-exchange.python.markdown (hunk 1 above)
Comment (reebhub) [reply Documentation/5.4/Raven.Documentation.Pages/client-api/session/cluster-transaction/compare-exchange.python.markdown:29]: done

File: Documentation/5.4/Raven.Documentation.Pages/client-api/session/cluster-transaction/compare-exchange.python.markdown [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+{NOTE/}
+
+{NOTE: }
+__Get multiple values__
+
+{CODE:python methods_2_sync@ClientApi\Session\ClusterTransaction\CompareExchange.py /}
+
+| Parameters   | Type       | Description               |
+|--------------|------------|---------------------------|
+| **keys**     | `str[]`    | Array of keys to retrieve |
```
[the remark this answers is not in the document]
Comment (reebhub) [reply Documentation/5.4/Raven.Documentation.Pages/client-api/session/cluster-transaction/compare-exchange.python.markdown:84]: done

File: Documentation/5.4/Raven.Documentation.Pages/client-api/session/cluster-transaction/compare-exchange.python.markdown (hunk 3 above)
Comment (reebhub) [reply Documentation/5.4/Raven.Documentation.Pages/client-api/session/cluster-transaction/compare-exchange.python.markdown:99]: done

File: Documentation/5.4/Raven.Documentation.Pages/client-api/session/cluster-transaction/compare-exchange.python.markdown (hunk 4 above)
Comment (reebhub) [reply Documentation/5.4/Raven.Documentation.Pages/client-api/session/cluster-transaction/compare-exchange.python.markdown:117]: done

File: Documentation/5.4/Raven.Documentation.Pages/client-api/session/cluster-transaction/compare-exchange.python.markdown (hunk 5 above)
Comment (reebhub) [reply Documentation/5.4/Raven.Documentation.Pages/client-api/session/cluster-transaction/compare-exchange.python.markdown:88]: done

File: Documentation/5.4/Raven.Documentation.Pages/client-api/session/cluster-transaction/compare-exchange.python.markdown (hunk 2 above)
Comment (reebhub) [reply Documentation/5.4/Raven.Documentation.Pages/client-api/session/cluster-transaction/compare-exchange.python.markdown:55]: edited the descriptions a bit

**Review by poissoncorp — APPROVED** [review]

**Pull request merged**